modin/test/test_docstring_urls.py

```diff
@@ -0,0 +1,49 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+from urllib.request import urlopen
+from urllib.error import HTTPError
+from concurrent.futures import ThreadPoolExecutor
+import pkgutil
+import importlib
+import pytest
+
+import modin.pandas
+
+
+@pytest.fixture
+def doc_urls(get_generated_doc_urls):
+    # ensure all docstring are generated - import _everything_ under 'modin.pandas'
+    for modinfo in pkgutil.walk_packages(modin.pandas.__path__, "modin.pandas."):
+        try:
+            importlib.import_module(modinfo.name)
+        except ModuleNotFoundError:
+            # some optional 3rd-party dep missing, ignore
+            pass
+    return sorted(get_generated_doc_urls())
+
+
+def test_all_urls_exist(doc_urls):
+    broken = []
+
+    def _test_url(url):
+        try:
+            with urlopen(url):
```

Comment:
Are you sure we really need to load all documentation URLs? Won't that get us banned for DDoSing pandas documentation web server?

Reply:
No idea, but I hope it won't.
I don't have any other good way of checking we're not generating some 404 URLs, though, except straight up checking them.

Reply:
Can we run the test on push rather than on PRs? That will reduce the amount of times it is run. We can create a separate test like we do with `ray-master`

Reply:
Great idea, moved.

Reply:
> Great idea, moved.

I see you moved call to `modin/test/test_docstring_urls.py` into `push.yml`. It runs `on: push` which according to [GitHub actions docs](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions) makes it run on every push to every branch. Not just push to a pull-request branch. It means that this test will run more often, not less often. Is it what you want to happen?

Reply:
huh, where should I have put it then?.. :)

Reply:
That's a good question, but `ci.yml` is better than `push.yml` because `ci` is run for PRs while `push` is run for every commit.